src/django_components/extensions/cache.py

```diff
@@ -15,7 +16,7 @@
 CACHE_KEY_PREFIX = "components:cache:"
 
 
-class ComponentCache(ComponentExtension.ExtensionClass):  # type: ignore
```

Comment:
Another minor issue with the Extension classes was that to define the inner class (now "ComponentConfig"), it had to subclass from

`ComponentExtension.ExtensionClass`

But Mypy doesn't like that, and it doesn't allow to use attributes when specifying bases for subclassing.

So now the base class of the inner class is part of public API as `ExtensionComponentConfig`, so that extensions can be created without this annoying error.